src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -1103,4 +1139,188 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
+                return 24;
+            case BSON_INT_TYPE_NAME:
+                return 10;
+            case BSON_BOOL_TYPE_NAME:
+                return 1;
+            case BSON_DECIMAL_TYPE_NAME:
+                return 34;
+            case BSON_DOUBLE_TYPE_NAME:
+                return 15;
+            case BSON_LONG_TYPE_NAME:
+                return 19;
+        }
+        throw new SQLException("unknown bson typeName: " + typeName);
+    }
+
+    protected Boolean getCaseSensitivity(String typeName) throws SQLException {
+        switch (typeName) {
+            case BSON_STRING_TYPE_NAME:
+            case BSON_REGEX_TYPE_NAME:
+            case BSON_SYMBOL_TYPE_NAME:
+            case BSON_JAVASCRIPT_TYPE_NAME:
+            case BSON_JAVASCRIPTWITHSCOPE_TYPE_NAME:
+                return true;
+            case BSON_BINDATA_TYPE_NAME:
+            case BSON_INT_TYPE_NAME:
+            case BSON_BOOL_TYPE_NAME:
+            case BSON_DATE_TYPE_NAME:
+            case BSON_DECIMAL_TYPE_NAME:
+            case BSON_DOUBLE_TYPE_NAME:
+            case BSON_LONG_TYPE_NAME:
+            case BSON_ARRAY_TYPE_NAME:
+            case BSON_OBJECT_TYPE_NAME:
+            case BSON_OBJECTID_TYPE_NAME:
+            case BSON_DBPOINTER_TYPE_NAME:
+            case BSON_MAXKEY_TYPE_NAME:
+            case BSON_MINKEY_TYPE_NAME:
+            case BSON_TIMESTAMP_TYPE_NAME:
+            case BSON_UNDEFINED_TYPE_NAME:
+            case BSON_BSON_TYPE_NAME:
+                return false;
+        }
+        throw new SQLException("unknown bson typeName: " + typeName);
+    }
+
+    protected Boolean getFixedPrecScale(String typeName) throws SQLException {
+        switch (typeName) {
+            case BSON_INT_TYPE_NAME:
+            case BSON_LONG_TYPE_NAME:
+                return true;
+            case BSON_BINDATA_TYPE_NAME:
+            case BSON_STRING_TYPE_NAME:
+            case BSON_BOOL_TYPE_NAME:
+            case BSON_DATE_TYPE_NAME:
+            case BSON_DECIMAL_TYPE_NAME:
+            case BSON_DOUBLE_TYPE_NAME:
+            case BSON_ARRAY_TYPE_NAME:
+            case BSON_OBJECT_TYPE_NAME:
+            case BSON_OBJECTID_TYPE_NAME:
+            case BSON_DBPOINTER_TYPE_NAME:
+            case BSON_JAVASCRIPT_TYPE_NAME:
+            case BSON_JAVASCRIPTWITHSCOPE_TYPE_NAME:
+            case BSON_MAXKEY_TYPE_NAME:
+            case BSON_MINKEY_TYPE_NAME:
+            case BSON_REGEX_TYPE_NAME:
+            case BSON_SYMBOL_TYPE_NAME:
+            case BSON_TIMESTAMP_TYPE_NAME:
+            case BSON_UNDEFINED_TYPE_NAME:
+            case BSON_BSON_TYPE_NAME:
+                return false;
+        }
+        throw new SQLException("unknown bson typeName: " + typeName);
+    }
+
+    protected Integer getMinScale(String typeName) throws SQLException {
+        switch (typeName) {
+            case BSON_DECIMAL_TYPE_NAME:
+                return 34;
+            case BSON_DOUBLE_TYPE_NAME:
+                return 15;
```

Comment:
For Double, it's a floating point decimal so I am not totally sure because it does not really apply, but certainly it can be < 15. I think 0 also makes sense.